spharpy/classes/ambisonics.py

```diff
@@ -0,0 +1,185 @@
+            half the sampling rate.
+        sampling_rate : double
+            Sampling rate in Hz
+        n_max : int
+            Maximum spherical harmonic order. Has to match the number of
+            coefficients, such that the number of coefficients
+            >= (n_max + 1) ** 2.
+        basis_type : str
+            Type of spherical harmonic basis, either ``'complex'`` or
+            ``'real'``. The default is ``'complex'``.
+        normalization : str
+            Normalization convention, either ``'n3d'``, ``'maxN'`` or
+            ``'sn3d'``. The default is ``'n3d'``.
+            (maxN is only supported up to 3rd order)
+        channel_convention : str
+            Channel ordering convention, either ``'acn'`` or ``'fuma'``.
+            The default is ``'acn'``.
+            (FuMa is only supported up to 3rd order)
+        phase_convention : str
+            Either 'condon-shortley', or 'ambix'.
+            TODO: discuss ...
+        n_samples : int, optional
+            Number of samples of the time signal. Required if domain is
+            ``'freq'``. The default is ``None``, which assumes an even number
+            of samples if the data is provided in the frequency domain.
+        domain : ``'time'``, ``'freq'``, optional
+            Domain of data. The default is ``'time'``
+        fft_norm : str, optional
+            The normalization of the Discrete Fourier Transform (DFT). Can be
+            ``'none'``, ``'unitary'``, ``'amplitude'``, ``'rms'``, ``'power'``,
+            or ``'psd'``. See :py:func:`~pyfar.dsp.fft.normalization` and [#]_
+            for more information. The default is ``'none'``, which is typically
+            used for energy signals, such as impulse responses.
+        comment : str
+            A comment related to `data`. The default is ``None``.
+
+        References
+        ----------
+        ..
+
+        """
+
+        if not data.shape[-2] >= (n_max + 1) ** 2:
+            raise ValueError('Data has to few sh coefficients '
+                             'for n_max = {n_max}.')
+
+        self._n_max = n_max
+
+        if basis_type == 'complex' and not is_complex:
+            raise ValueError('Data are real-valued while '
+                             'spherical harmonics bases are complex-valued.')
+
+        self._basis_type = basis_type
+        self._phase_convention = phase_convention
+
+        if normalization in ['sn3d', 'n3d', 'maxN']:
+            self._normalization = normalization
+        else:
+            raise ValueError("Invalid normalization, has to be 'sn3d', "
+                             f"'n3d', or 'maxN, but is {normalization}")
+
+        if channel_convention in ['acn', 'fuma']:
+            self._channel_convention = channel_convention
+        else:
+            raise ValueError("Invalid channel convention, has to be 'acn' "
+                             f"or 'fuma', but is {channel_convention}")
```

Comment:
I would copy getters and setters from the SH class once this is finished!

Reply:
s.o.